packages/main/src/FileUploader.js

```diff
@@ -0,0 +1,318 @@
+		return true;
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return FileUploaderCss;
+	}
+
+	static get template() {
+		return FileUploaderTemplate;
+	}
+
+	constructor() {
+		super();
+		if (this._canUseNativeFormSupport) {
+			this._internals = this.attachInternals();
+		}
+
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	_onmouseover() {
+		this.content.forEach(item => {
+			item.classList.add("ui5_hovered");
+		});
+	}
+
+	_onmouseout() {
+		this.content.forEach(item => {
+			item.classList.remove("ui5_hovered");
+		});
+	}
+
+	/**
+	 * FileList of all selected files.
+	 * @readonly
+	 * @type { FileList }
+	 * @public
+	 */
+	get files() {
+		if (this._input) {
+			return this._input.files;
+		}
+
+		return FileUploader._emptyFilesList;
+	}
+
+	onBeforeRendering() {
+		this._enableFormSupport();
+	}
+
+	_enableFormSupport() {
+		const FormSupport = getFeature("FormSupport");
+
+		if (FormSupport) {
+			if (this._canUseNativeFormSupport) {
+				this._setFormValue();
+			} else {
+				FormSupport.syncNativeFileInput(
+					this,
+					(element, nativeInput) => {
+						nativeInput.disabled = element.disabled;
+					},
+					this._onChange.bind(this)
+				);
+			}
+		} else if (this.name) {
+			console.warn(`In order for the "name" property to have effect, you should also: import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";`); // eslint-disable-line
+		}
+	}
+
+	_onChange(event) {
+		this._updateValue(event.target.files);
+		this.fireEvent("change", {
+			files: event.target.files,
+		});
+	}
+
+	_updateValue(files) {
+		this.value = Array.from(files).reduce((acc, currFile) => {
+			return `${acc}"${currFile.name}" `;
+		}, "");
+	}
+
+	_setFormValue() {
+		const formData = new FormData();
+
+		for (let i = 0; i < this.files.length; i++) {
+			formData.append(this.name, this.files[i]);
+		}
+
+		this._internals.setFormValue(formData);
+	}
+
+	/**
+	 * in case when ui5-file-uploader is not placed in the DOM, return empty FileList, like native input would do
+	 * @private
+	 */
+	static get _emptyFilesList() {
+		if (!this.emptyInput) {
+			this.emptyInput = document.createElement("input");
+			this.emptyInput.type = "file";
+		}
+		return this.emptyInput.files;
+	}
+
+	get browseText() {
+		return this.i18nBundle.getText(FILEUPLOAD_BROWSE);
+	}
+
+	get _canUseNativeFormSupport() {
+		return !!this.attachInternals;
+	}
+
+	get _keepInputInShadowDOM() {
+		// only put input in the light dom when ui5-file-uploader is placed inside form and there is no support for form elements
+		return this._canUseNativeFormSupport || !this.name;
+	}
+
+	get _input() {
+		return this.shadowRoot.querySelector("input[type=file]") || this.querySelector("input[type=file][data-ui5-form-support]");
+	}
+
+	/**
+	 * Determines input helper type in forms.
+	 * @private
+	 */
+	get _type() {
+		return "file";
+	}
+
+	static async onDefine() {
```

Comment:
is it onDefine on purpose?

Reply:
Yes, we changed the method name and the way it works